_posts/2015-07-21-build.md

```diff
@@ -0,0 +1,55 @@
+# Servo build system overview
+
+[Servo](https://github.com/servo/servo) is a new web browser engine. It is one of the largest Rust-based projects, but the total Rust code is still dwarfed by the size of the code provided in native C and C++ libraries. This post is an overview of how we have structured our build system in order to integrate the Cargo build system, with its "many small and distributed dependencies" model with our needs to provide many additional features not often found in smaller Rust-only projects.
+
+## Mach
+
+[Mach](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/mach) is a python driver program that provides a frontend to Servo's build that both reduces the number of steps to build and integrates our different testing tools into a single frontend harness. Similar to its purpose in the Firefox build, we use it to centralize and simplify the number of commands that a developer has to perform.
+
+### mach bootstrap
+
+The steps that mach will handle before issuing a normal `cargo build` command are:
+* Downloading the correct versions of the cargo and rustc tools. Servo uses most of the unstable features in Rust, which change pretty frequently. We also test the edges of feature compatibility and so are the first ones to notice many changes that did not at first seem as if they would break anyone. Further, we build a custom version of the tools that additionally supports cross-compilation targeting Android (and ARM in the near future). A random local install of the Rust toolchain is pretty unlikely to work with Servo.
```

Comment:
> Servo uses most of the unstable features in Rust

I'd say this has changed. We use some of the _most_ unstable features of Rust, but actually our unstable usage is pretty low other than plugins and some APIs here and there.